db/migrate/20171029155754_move_applicants_to_new_users.rb

```diff
@@ -0,0 +1,182 @@
+            ('CS Degree', 'application', true),
+            ('Worked as Dev', 'application', true)
+      RETURNING id
+    SQL
+
+    field_ids = field_ids.map { |field_id| field_id['id'] }
+
+    # Add choices for the applicable fields
+    execute <<-SQL
+      INSERT INTO field_choices(name, field_id)
+      VALUES('Chicago', #{field_ids[1]}),
+            ('London', #{field_ids[1]}),
+            ('Los Angeles', #{field_ids[1]}),
+            ('Developer', #{field_ids[2]}),
+            ('Designer', #{field_ids[2]}),
+            ('10+ years', #{field_ids[3]}),
+            ('2-9 years', #{field_ids[3]}),
+            ('2 years or less', #{field_ids[3]}),
+            ('None', #{field_ids[7]}),
+            ('App Academy', #{field_ids[7]}),
+            ('Bitmaker Labs', #{field_ids[7]}),
+            ('Dev Bootcamp', #{field_ids[7]}),
+            ('General Assembly', #{field_ids[7]}),
+            ('Google SOC', #{field_ids[7]}),
+            ('HackBright', #{field_ids[7]}),
+            ('Hacker School', #{field_ids[7]}),
+            ('Launch Academy', #{field_ids[7]}),
+            ('Makers Academy', #{field_ids[7]}),
+            ('Mobile Makers', #{field_ids[7]}),
+            ('Start Up Institute', #{field_ids[7]}),
+            ('Starter League', #{field_ids[7]}),
+            ('None', #{field_ids[8]}),
+            ('Yes', #{field_ids[8]}),
+            ('No', #{field_ids[8]}),
+            ('None', #{field_ids[9]}),
+            ('Yes', #{field_ids[9]}),
+            ('No', #{field_ids[9]}),
+            ('None', #{field_ids[10]}),
+            ('Yes', #{field_ids[10]}),
+            ('No', #{field_ids[10]})
+    SQL
+
+    field_map = {
+      website: field_ids[0],
+      location: field_ids[1],
+      discipline: field_ids[2],
+      skill_level: field_ids[3],
+      about: field_ids[4],
+      software_interest: field_ids[5],
+      reason: field_ids[6],
+      codeschool: field_ids[7],
+      college_degree: field_ids[8],
+      cs_degree: field_ids[9],
+      worked_as_dev: field_ids[10]
+    }
+
+    applicants = ::Applicant.find_by_sql <<-SQL
+      SELECT applicants.*, new_users.id AS new_user_id, filled_forms.id AS filled_form_id
+      FROM applicants
+      JOIN new_users ON applicants.email = new_users.email
+      JOIN filled_forms ON new_users.id = filled_forms.applicant_id
+    SQL
+
+    answers = applicants.map do |applicant|
+      [
+        {
+          user_id: applicant.new_user_id,
+          field_id: field_map[:website],
+          filled_form_id: applicant.filled_form_id,
+          field_choice_id: nil,
+          answer_text: applicant.url
+        },
+        {
+          user_id: applicant.new_user_id,
+          field_id: field_map[:location],
+          filled_form_id: applicant.filled_form_id,
+          field_choice_id: execute("SELECT id FROM field_choices WHERE field_id = #{field_map[:location]} AND name ILIKE '#{applicant.location}'").first[:id],
```

Comment:
This seems to always return nil. Need to investigate why.

Reply:
Investigated... seems like it was below on `skill_level` instead though. We had `10+ years`, `2-9 years`, and `2 years or less` for the field_choice.name rather than the actual values that existed on applicants (`craftsmen`, `resident`, `student`

Reply:
Ah yeah that'll do it.  I thought they were all returning nil, nice to know it was just one of them.